### ISSUE TITLE
Graphical week view

### ISSUE THREAD

**Issue by raxod502** [kiwanami/emacs-calfw#112]
I would love to do my calendaring in Emacs using calfw. However, I rely heavily on a graphical week view that shows me events as blocks of the appropriate size, rather than just as list items. Is there any possibility for calfw to support such a week view, like Apple/Google Calendar?

![image](https://user-images.githubusercontent.com/6559064/40880101-0b322ed6-6668-11e8-9083-8aa910fd3f60.png)

**Comment by fbob** [comment]
+1
It could be indeed a great feature.
But I'm not sure if it's possible at present time.

**Comment by lgmoneda** [comment]
I'm also currently looking for a solution like that for org-mode: being able to see tasks as time blocks, also because it makes clear the free time blocks.

Have you found a solution, @raxod502? I'd love to see it as a `calfw` feature.

**Comment by raxod502** [comment]
I'm developing my own Emacs calendar interface to solve this problem. It will be designed to integrate well with [vdirsyncer](https://github.com/pimutils/vdirsyncer).

The code is only at a very preliminary stage, and doesn't accomplish anything useful yet. If you want to follow the project, it is at https://github.com/raxod502/ishikk.

**Comment by fbob** [comment]
great ! you should add a screenshot on your github page
best wishes for your project

**Comment by raxod502** [comment]
Of course—although obviously that is in the future since as I've said above the code doesn't actually do anything yet.

**Comment by lgmoneda** [comment]
@raxod502 I think the [org-timeline](https://github.com/Fuco1/org-timeline/) is mildly related to what you want to build. I would really enjoy a weekly-transposed-with-task-titles version of org-timeline!

**Comment by dudelson** [comment]
This feature is a must-have for me as well. @fbob @kiwanami what are the technical limitations to something like this? I'd be willing to take a crack at a PR myself, but I am not very knowledgeable about emacs lisp, so some direction would be appreciated :+1:

**Comment by novoid** [comment]
From my perspective, #22 is related to similar requirements: a more horizontal layout in contrast to the clearly vertical oriented layout which is not compatible with longer text per appointment.

**Comment by ml729** [comment]
Hello, I'm a bit late to the party, but I implemented a feature like this here: https://github.com/ml729/calfw-blocks

**Comment by novoid** [comment]
> Hello, I'm a bit late to the party, but I implemented a feature like this here: https://github.com/ml729/calfw-blocks

Wouldn't you agree that transposing the table (Weekdays are the rows, events in the columns) would be a much better choice since the titles/descriptions doesn't have to be truncated that often?

See #22 for more details on that.

**Comment by ml729** [comment]
I like your design @novoid . For shorter events (like, less than 2 hours) I think truncation would still occur in the transposed view (e.g. you have a 160 char wide buffer and you're displaying 8 hours at a time, then you only get 20 chars of horizontal space for a 1 hour event). For longer events, yes you would have less truncation/line wrapping. Unfortunately, I currently don't plan on implementing it, but PRs are welcome! Also, if you want less truncation, my package does have views with fewer days, where each event gets more horizontal space.

By the way, there is a PR of org-timeline that does implement a nice transposed view, but sacrifices titles in the process. Fuco1/org-timeline#19
(There is also an org-mode extension for Sublime Text that has a similar display in its org-agenda. https://github.com/ihdavids/orgextended)

If you are only interested in having the titles and not visually displaying time/duration, I think this could be implemented more cleanly within the org-agenda view rather than calfw. The events are already listed by day, vertically, you just have to transform it into a two column format.

**Comment by ml729** [comment]
@novoid, I ended up implementing the view in your image (https://github.com/kiwanami/emacs-calfw/issues/22) as part of calfw-blocks.

**Comment by Gleek** [comment]
It's been 8 years, since I last 👍 'd this issue. 
But I think I have something pretty close and I've been using it for a month. 
Dropping this here in case people had subscribed this thread for further updates.
https://github.com/Gleek/org-timegrid

https://github.com/user-attachments/assets/2e1df1c3-d0f0-419d-b5f2-66d876753872
